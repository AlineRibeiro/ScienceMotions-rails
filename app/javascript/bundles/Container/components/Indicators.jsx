import PropTypes from 'prop-types';
import React from 'react';

export default class Indicators extends React.Component {
  static propTypes = {
    name: PropTypes.string.isRequired, // this is passed from the Rails view
  };

  /**
   * @param props - Comes from your rails view.
   */
  constructor(props) {
    super(props);

    // How to set initial state in ES6 className syntax
    // https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-className
    this.state = { name: this.props.name };
  }

  updateName = (name) => {
    this.setState({ name });
  };

  render() {
    return (
        <div class="table">indicators react renders
        </div>     
    );
  }
}
