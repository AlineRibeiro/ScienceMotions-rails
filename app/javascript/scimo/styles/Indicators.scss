@import url("https://fonts.googleapis.com/css?family=Fugaz+One");
@import url("https://fonts.googleapis.com/css?family=Space+Mono");

.indicators {
  padding: 0;
  font-family: "Space mono", sans-serif;
  text-align: justify;
}

.redlitmus {
  width: 250px;
  height: 75px;
  background-color: #db6066;
  box-shadow: 10px 10px 5px grey;
}

.bluelitmus {
  width: 250px;
  height: 75px;
  background-color: #608fdb;
  box-shadow: 10px 10px 5px grey;
}

.red:hover {
  -webkit-animation-name: turnred;
  /* Safari 4.0 - 8.0 */
  -webkit-animation-duration: 4s;
  /* Safari 4.0 - 8.0 */
}

@-webkit-keyframes turnred {
  from {
    background-color: #608fdb;
  }
  to {
    background-color: #db6066;
  }
}

.blu:hover {
  -webkit-animation-name: turnblue;
  /* Safari 4.0 - 8.0 */
  -webkit-animation-duration: 4s;
  /* Safari 4.0 - 8.0 */
}

@-webkit-keyframes turnblue {
  from {
    background-color: #db6066;
  }
  to {
    background-color: #608fdb;
  }
}
