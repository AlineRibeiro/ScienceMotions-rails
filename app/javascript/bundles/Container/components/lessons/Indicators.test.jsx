import React from 'react';
import renderer from 'react-test-renderer';
import Indicators from './Indicators';

describe('Indicators', () => {
  it('renders correctly when there are no items', () => {
    const rendered = renderer.create(<Indicators />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
})
