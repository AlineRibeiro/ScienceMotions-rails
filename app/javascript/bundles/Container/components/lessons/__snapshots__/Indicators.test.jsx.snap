// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicators renders correctly when there are no items 1`] = `
Array [
  <section
    className="lesson-container indicators"
    id="visual"
  >
    <div
      className="col-sm-1"
    />
    <div
      className="col-sm-3"
    >
      <p>
        Acidic Solution
      </p>
      <p
        className="animated infinite pulse"
      >
        [H+] &gt; [OH-]
      </p>
      <div
        className="redlitmus animated bounceIn"
      >
        Red litmus - stays red
      </div>
      <br />
      <div
        className="bluelitmus red animated bounceIn"
      >
        Blue litmus - turns red
      </div>
    </div>
    <div
      className="col-sm-1"
    />
    <div
      className="col-sm-3"
    >
      <p>
        Neutral Solution
      </p>
      <p
        className="animated infinite pulse"
      >
        [H+]=[OH-]
      </p>
      <div
        className="redlitmus animated bounceIn"
      >
        Red litmus - stays red
      </div>
      <br />
      <div
        className="bluelitmus animated bounceIn"
      >
        Blue litmus - stays blue
      </div>
    </div>
    <div
      className="col-sm-1"
    />
    <div
      className="col-sm-3"
    >
      <p>
        Basic Solution
      </p>
      <p
        className="animated infinite pulse"
      >
        [H+] &lt; [OH-] 
      </p>
      <div
        className="redlitmus blu animated bounceIn"
      >
        Red litmus - turns blue
      </div>
      <br />
      <div
        className="bluelitmus animated bounceIn"
      >
        Blue litmus - stays blue
      </div>
    </div>
  </section>,
  <section
    id="text"
  >
    <h1>
      Indicators
    </h1>
    <p>
      When an acid is dissolved in water we get an 
      <span
        id="text-1"
      >
        acidic
      </span>
       solution. When a base dissolves in water it is an alkali and makes an 
      <span
        id="text-2"
      >
        alkaline
      </span>
       solution. If a solution is neither acidic nor alkaline it is 
      <span
        id="text-3"
      >
        neutral
      </span>
      . Pure water is neutral, and so is paraffin.
    </p>
    <p>
      Indicators are substances that change colour when they are added to acidic or alkaline solutions. Litmus, phenolphthalein, and methyl orange are all indicators that are commonly used in the laboratory.
    </p>
    <p>
      <span
        id="text-1"
      >
        Litmus
      </span>
    </p>
    <p>
      Litmus indicator solution turns red in acidic solutions and blue in alkaline solutions - and it turns purple in neutral solutions.
    </p>
    <p>
      <span
        id="text-2"
      >
        Litmus paper
      </span>
       is usually more reliable, and comes as red litmus paper and blue litmus paper. The table shows the colour changes it can make.
    </p>
    <p>
      The table above shows the colour changes it can make.
    </p>
  </section>,
]
`;
